node/flows.re

```diff
@@ -152,14 +152,94 @@ let request_previous_blocks = (state: Node_state.t, block) =>
     request_protocol_snapshot();
   };
 
-let try_to_produce_block = (state, update_state) => {
+/** Returns true if it is time to start hashing a new state root.
+ */
+let is_new_state_root_epoch = (last_state_root_update, current_time) => {
+  /** The state root epoch determines how often the block producer
+    starts hashing a new state root. It is closely related to the
+    finality period (see notes in [try_to_produce_block]).
+
+    The faster hashing is, the shorter this time should be (resulting
+    in faster finality).
+ */
+  let state_root_hash_epoch = 60.0;
+  /** to prevent changing the validator just because of network jittering
+    this introduce a delay between can receive a block with new state
+    1s choosen here but any reasonable time will make it */
+  let avoid_jitter = 1.0;
+  current_time
+  -. last_state_root_update
+  -. avoid_jitter >= state_root_hash_epoch;
+};
+
+let try_hash_new_state =
+    (
+      ~state: Node_state.t,
+      ~current_time,
+      ~update_state: Node_state.t => Node_state.t,
+    ) => {
+  let (_, _, last_state_root_update) = List.hd(state.state_root_hash_list);
+  // If it's a new epoch then start hashing the current state
+  if (is_new_state_root_epoch(last_state_root_update, current_time)) {
+    open Lwt.Infix;
+    // Infix operators used because `let.async` transforms whole function
+    // into Lwt monad. Instead, we use infix and ignore the result because
+    // we only care about the side effects, not the result.
+    let _ =
+      // Lwt_domain.detach causes the function to run in a separate thread.
+      Lwt_domain.detach(
+        (s: Node_state.t) =>
+          (
+            Protocol.hash(s.protocol),
+            Validators.hash(s.protocol.validators),
+          ),
+        state,
+      )
+      // When the promise resolves, the following callback is run in main thread.
+      >|= (
```

Comment:
Don't use infix operators, refer to existing code on how to do it.

Reply:
Addressed here: https://github.com/marigold-dev/deku/pull/195/commits/2ebfd66460933c061c84bdb66f6dce484547a1ff